src/main.rs

```diff
@@ -378,6 +363,18 @@ async fn run(miniserve_config: MiniserveConfig) -> Result<(), ContextualError> {
         .map_err(|e| ContextualError::IoError("".to_owned(), e))
 }
 
+// Allows us to set low-level socket options:
```

Comment:
Let's use proper docstrings (`///`) for documenting functions.

Reply:
Okay.

What do you think about this overall?
I want to settle this to push #422 fix

Reply:
I really like it. I'm just skeptical about cross-platform support which is very important to me. This needs to work the same on all operating systems (which arguably currently it doesn't really, especially on Windows).